tests/NRedisStack.Tests/SkipIfRedisVersionGte.cs

```diff
@@ -0,0 +1,32 @@
+using Xunit;
+using StackExchange.Redis;
+
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+public class SkipIfRedisVersionGteAttribute : FactAttribute
```

Comment:
I like this approach _a lot_, my only comment is that it looks like there's an opportunity to condense this down into one attribute `SkipIfRedisVersionAttribute`, with version and comparison type. As it stands right now there's a `gte` and `lt` so it feels somewhat incomplete

Reply:
somthing like this?
```csharp
using Xunit;
using StackExchange.Redis;

public enum Comparison
{
    LessThan,
    LessThanOrEqual,
    GreaterThan,
    GreaterThanOrEqual,
    Equal
}

[AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
public class SkipIfRedisVersionAttribute : FactAttribute
{
    private readonly string _targetVersion;
    private readonly Comparison _comparison;
    private readonly string DefaultRedisConnectionString = Environment.GetEnvironmentVariable("REDIS") ?? "localhost:6379";

    public SkipIfRedisVersionAttribute(string targetVersion, Comparison comparison = Comparison.GreaterThanOrEqual)
    {
        _targetVersion = targetVersion;
        _comparison = comparison;
    }

    public override string Skip
    {
        get
        {
            using (var connection = ConnectionMultiplexer.Connect(DefaultRedisConnectionString))
            {
                var serverVersion = connection.GetServer(connection.GetEndPoints()[0]).Version;
                var targetVersion = new Version(_targetVersion);
                int comparisonResult = serverVersion.CompareTo(targetVersion);

                switch (_comparison)
                {
                    case Comparison.LessThan:
                        if (comparisonResult < 0)
                            return $"Test skipped because Redis server version ({serverVersion}) is less than {_targetVersion}.";
                        break;
                    case Comparison.LessThanOrEqual:
                        if (comparisonResult <= 0)
                            return $"Test skipped because Redis server version ({serverVersion}) is less than or equal to {_targetVersion}.";
                        break;
                    case Comparison.GreaterThan:
                        if (comparisonResult > 0)
                            return $"Test skipped because Redis server version ({serverVersion}) is greater than {_targetVersion}.";
                        break;
                    case Comparison.GreaterThanOrEqual:
                        if (comparisonResult >= 0)
                            return $"Test skipped because Redis server version ({serverVersion}) is greater than or equal to {_targetVersion}.";
                        break;
                    case Comparison.Equal:
                        if (comparisonResult == 0)
                            return $"Test skipped because Redis server version ({serverVersion}) is equal to {_targetVersion}.";
                        break;
                }

                return null;
            }
        }
    }
}

```

Reply:
yeah, exactly, I think I would switch the order of the parameters in the constructor that way it will read like:

```cs
[SkipIfRedisVersion(Comparison.LessThan,"7.2.242")]
```

Reply:
also default should probably be less than? (presumably this will be mostly used for new features.)

Reply:
Over time we will need:
LessThan
GreaterThan

You can see these in other clients.